src/base/static/js/views/app-view.js

```diff
@@ -846,52 +699,100 @@ var Shareabouts = Shareabouts || {};
           }
         }
 
-        // Focus the one we're looking
         model.trigger('focus');
-
-        if (model.get("story")) {
-          if (!model.get("story").spotlight) $("#spotlight-place-mask").remove();
-          self.isStoryActive = true;
-          self.setStoryLayerVisibility(model);
-        } else if (self.isStoryActive) {
+        
+        if (!model.get("story") && self.isStoryActive) {
           self.isStoryActive = false;
           self.restoreDefaultLayerVisibility();
-        } else {
-          self.isStoryActive = false;
         }
       };
 
-      onPlaceNotFound = function() {
+      onNotFound = function() {
         self.options.router.navigate('/');
+        return;
       };
 
-      // If we get a PlaceModel then show it immediately.
-      if (model instanceof S.PlaceModel) {
-        onPlaceFound(model);
-        return;
-      }
+      searchLoadedCollections = function(collections, property, type) {
+        var found = false,
+        searchTerm = {};
+        searchTerm[property] = args.modelId;
+        _.find(collections, function(collection, datasetId) {
+          var model = collection.where(searchTerm);
+          if (model.length === 1) {
+            found = true;
+            onFound(model[0], type, datasetId);
+            return;
+          }
+        });
 
-      // Otherwise, assume we have a model ID.
-      modelId = model;
-      model = this.places[datasetId].get(modelId);
+        return found;
+      };
 
-      // If the model was found in the places, go ahead and use it.
-      if (model) {
-        onPlaceFound(model);
+      bindCollectionsListeners = function(collections, property, type, finalCollection) {
+        var numCollections = _.keys(collections).length,
+        i = 0,
+        searchTerm = {};
+        searchTerm[property] = args.modelId;
+        _.each(collections, function(collection, datasetId) {
+          collection.on("sync", function(syncedCollection) {
+            i++;
+            var model = syncedCollection.where(searchTerm);
+            if (model.length === 1) {
+              foundInCallback = true;
+              onFound(model[0], type, datasetId);
+            } else if (numCollections === i && finalCollection && !foundInCallback) {
```

Comment:
Subtle issue, but instead of traversing all collections with a `.each` and tracking a index i, we can use a regular for loop (ie `for (i = 0; i < collections.length; i++) {..}`), and just add a check after the for loop has been traversed (ie `if (!foundInCallback && isFinalCollection) {..}`).